text/0048-schema-annotations.md

```diff
@@ -0,0 +1,139 @@
+# Annotations for Schemas 
+
+## Related issues and PRs
+
+- Reference Issues: 
+- Implementation PR(s): 
+
+## Timeline
+
+- Started: 2024-02-05
+
+## Summary
+
+Like Cedar policies, users may want to associate arbitrary, machine readable metadata with Schema objects.
+We solved this problem in Cedar Policies by allowing for *annotations*: arbitrary key/value pairs that are attachable to policies.
+This could be extended to Cedar Schemas, allowing users to attach attributes an entity type, common type, and action definitions.
+
+
+## Basic example
+
+1. Doc comments
+```
+@doc("This entity defines our central user type")
```

Comment:
You have a number here as if this is the first of several examples, but it's the only one.

For doc comments, it makes more sense to me to define an opinionated literate format. This is what Rust, Java, etc. all do, and it's nicer to read and perhaps more natural to write (per your Alternative "Doc strings as comments"). But we could still _also_ have arbitrary annotations to handle other use-cases (which it would be nice to see, here).